crates/router/src/connector/gpayments/gpayments_types.rs

```diff
@@ -122,3 +125,127 @@ pub struct GpaymentsPreAuthenticationResponse {
     #[serde(rename = "threeDSServerTransID")]
     pub three_ds_server_trans_id: String,
 }
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct GpaymentsAuthenticationRequest {
+    pub acct_number: CardNumber,
+    pub authentication_ind: String,
+    pub browser_info_collected: BrowserInfoCollected,
+    pub card_expiry_date: String,
+    #[serde(rename = "notificationURL")]
+    pub notification_url: String,
+    pub merchant_id: String,
+    #[serde(rename = "threeDSCompInd")]
+    pub three_ds_comp_ind: ThreeDsCompletionIndicator,
+    pub message_category: String,
+    pub purchase_amount: String,
+    pub purchase_date: String,
+    #[serde(rename = "threeDSServerTransID")]
+    pub three_ds_server_trans_id: String,
+}
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct BrowserInfoCollected {
+    pub browser_accept_header: Option<String>,
+    pub browser_color_depth: Option<String>,
+    #[serde(rename = "browserIP")]
+    pub browser_ip: Option<Secret<String, common_utils::pii::IpAddress>>,
+    pub browser_javascript_enabled: Option<bool>,
+    pub browser_java_enabled: Option<bool>,
+    pub browser_language: Option<String>,
+    pub browser_screen_height: Option<String>,
+    pub browser_screen_width: Option<String>,
+    #[serde(rename = "browserTZ")]
+    pub browser_tz: Option<String>,
+    pub browser_user_agent: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum AuthenticationInd {
+    #[serde(rename = "01")]
+    PaymentTransaction,
+    #[serde(rename = "02")]
+    RecurringTransaction,
+    #[serde(rename = "03")]
+    InstalmentTransaction,
+    #[serde(rename = "04")]
+    AddCard,
+    #[serde(rename = "05")]
+    MaintainCard,
+    #[serde(rename = "06")]
+    CardholderVerification,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum GpaymentsAuthenticationResponse {
+    Success(Box<GpaymentsAuthenticationSuccessResponse>),
+    Error(Box<GpaymentsErrorResponse>),
+}
```

Comment:
This untagged enum is not required since Gpayments will always return `GpaymentsAuthenticationSuccessResponse` with 200 status code.